controllers/operatorpolicy_controller.go

```diff
@@ -134,9 +147,74 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// FUTURE: more resource checks
 
+	// handle catalogsource
+	if err := r.handleCatalogSource(ctx, policy, subscription); err != nil {
+		OpLog.Error(err, "Error handling CatalogSource")
+
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCatalogSource(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	subscription *operatorv1alpha1.Subscription,
+) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	catalogName := subscription.Spec.CatalogSource
+	catalogNS := subscription.Spec.CatalogSourceNamespace
```

Comment:
I think we need to guard against the situation where the subscription is nil here.

Reply:
Would the subscription ever be nil? As far as I can tell, `handleSubscription` only returns a nil subscription with an error, which would return before it can hit the `handleCatalogSource` block.

Reply:
Right now, it shouldn't ever be nil when it gets here. But things can change in the future, and if it does get here as `nil`, I think the controller would panic. I'd feel better to be on the safe side.